diff-hl.el

```diff
@@ -1442,19 +1442,25 @@ CONTEXT-LINES is the size of the unified diff context, defaults to 0."
    ((eq backend 'Git)
     (vc-git--rev-parse diff-hl-reference-revision))
    ((eq backend 'Hg)
-    (with-temp-buffer
-      (vc-hg-command (current-buffer) 0 nil
-                     "identify" "-r" diff-hl-reference-revision
-                     "-i")
-      (goto-char (point-min))
-      (buffer-substring-no-properties (point) (line-end-position))))
+    ;;  Preserve the potentially buffer-local `diff-hl-reference-revision`
+    ;;  inside the `with-temp-buffer`.
+    (let ((original-diff-hl-reference-revision diff-hl-reference-revision))
```

Comment:
Let's just call it `reference-revision`?

Reply:
Done.